### ISSUE TITLE
PD Upscaler - Flashing lights in some specular textures

### ISSUE THREAD

**Issue by DaveACG** [praydog/REFramework#673]
All RE Engine games (RE7/2/3/4) to some extent show this issue when using the PD Upscaler, DLSS or FSR. 

There are general limitations the plugin has when dealing with low-res buffer effects that are likely a function of not having these reconstruction methods baked into the engine from the start, such as specular shimmer when using motion blur/lens distortion, but these can be easily rectified by just disabling those graphical options in the game. This artifact however has no equivalent graphical setting that can control it, as well it occurs without any camera motion.

[Video of it in action in RE3 shown here.](https://youtu.be/3inxd7_EqtQ)

**Comment by TommyCreo21** [comment]
Yeah, I noticed this too and found it very distracting. If you enable "Allow Engine TAA", it gets rid of the shimmering. It does however also make the image significantly more blurry, which kinda defeats the whole point of using the upscaler.

**Comment by praydog** [comment]
I think I can fix this by removing the jitter in the post process effect pass. I'll have to try it.

**Comment by TommyCreo21** [comment]
While I can live without bloom, it would definitely be nice to have again. Doing upscaling before the post processing sounds like the ideal solution if possible.

Btw, thank you for your work praydog. This really enhances the game. RE Engine is great in many ways. It's just a shame that their AA has always been lackluster, and that they refuse to implement DLSS themselves.

**Comment by Johnnydana** [comment]
In the meantime, I have two solutions for you, use reshade to add high quality bloom back, and use VRtoolkit to add sharpness back if you set "image quality" down to 80% to help performance / Vram usage. Not as ideal as a non glitchy DLSS but you know.

**Comment by TommyCreo21** [comment]
Well, Reshade doesn't work with upscaler version of RE Framework, but Nvidia Freestyle does, and you can add reshade shaders to that. I did try and add Bloom via that, but haven't found a Bloom shader I like yet.

**Comment by JoHien** [comment]
> I think what Tommy means is, that Reshade doesn't work anymore as soon as you tick the "Upscale" checkbox under "Debug". I tried it with renaming dxgi.dll to d3d12.dll like you suggested but no dice. If you got it working, could you give some more insight how, please. Thanks!

So changing the reshade file from dxgi.dll to d3d12.dll does work. The thing is that you can only open one window at a time. For example, if you want to open reshade window, you would need to minimize REFramework. Same goes the other way.

**Comment by DHYCIX** [comment]
Unfortunately this isn't the case on my end. I can open the Reshade Window at the same time as the reframework one. It just doesn't display at all as soon as I tick the "Upscale" checkbox. When I press the Home key to open the Reshade window, I can tell it tries to open because the mouse pointer changes, but it's like it's "behind" the upscaled image and doesn't affect it at all.

**Comment by JoHien** [comment]
> Unfortunately this isn't the case on my end. I can open the Reshade Window at the same time as the reframework one. It just doesn't display at all as soon as I tick the "Upscale" checkbox. When I press the Home key to open the Reshade window, I can tell it tries to open because the mouse pointer changes, but it's like it's "behind" the upscaled image and doesn't affect it at all.

That's odd. Are you using dlsstweaks by chance? If you are, you may need to change dlsstweaks dxgi.dll to another name. Here's a video of when it's working correctly.
[DLSS + Reshade Enable](https://youtu.be/rUGLcODHGmw)

**Comment by DHYCIX** [comment]
Thank you for your insight. I'm using FSR as I'm running the game on an AMD GPU. Maybe that's the root of the issue. There are 2 .dll's that are neccessary for it to work but I can't rename them as reframework won't be able to find them anymore then.

I also tried Xess but it's crashing with the current build, so I can't say if the issue persists there, too.

**Comment by JoHien** [comment]
> Thank you for your insight. I'm using FSR as I'm running the game on an AMD GPU. Maybe that's the root of the issue. There are 2 .dll's that are neccessary for it to work but I can't rename them as reframework won't be able to find them anymore then.
> 
> I also tried Xess but it's crashing with the current build, so I can't say if the issue persists there, too.

I tried FSR2 and XeSS and it works with reshade too. I did a fresh install of reshade so maybe that's why? Kinda hard to pinpoint the issue. This how my game folder looks like.
[Game Folder for RE3](https://i.imgur.com/Y8flWzg.png)

**Comment by DHYCIX** [comment]
I just found out by chance I was on an older release. I'm sorry, Github is quite hard to navigate for amateurs like me, so I just couldn't find out how to get PD's nightly releases before. Thanks for troubleshooting with me, though. Without you, I wouldn't be able to enjoy this now. Everythink works. Yay! :)

**Comment by TeTr-a** [comment]
Hijacking this to mention that the newest update to the pdupscaler has made great improvements to the flickering caused by bloom. It's far from perfect, but it's much better now.

**Comment by TommyCreo21** [comment]
> Hijacking this to mention that the newest update to the pdupscaler has made great improvements to the flickering caused by bloom. It's far from perfect, but it's much better now.

Huh. In which game? I'm still seeing a lot of flickering in RE2, which seems to have it the worst.

**Comment by TommyCreo21** [comment]
Tried the previous build as well. Still shimmering in RE2. RE4 generally has a lot less shimmering. But haven't tried turning on Bloom again in RE4 since updating to build 440.

**Comment by DaveACG** [comment]
> Hijacking this to mention that the newest update to the pdupscaler has made great improvements to the flickering caused by bloom. It's far from perfect, but it's much better now.

Yeah I heard that from another forum, but after testing I see no change yet. Tried RE4 and RE3 (RE3 is really the best case for diagnosing this) with several of the most recent dev builds, don't see any difference. I don't see any notes in the releases that specifically pertain to this either.

_Maybe_ some improvement with how specular surfaces interact with depth of field, perhaps less there. But the flashbulbs still go off with bloom.

**Comment by Yudoken** [comment]
Any update  about the bloom issues? Currently playing through Resident Evil 4 with DLSS and the flashing lights while bloom was on wasn't that big of a deal until the underground laboratory section, the bloom issue now is just too much and I had to disable bloom to "fix" it. I'm using the pd-upscaler release #493

https://user-images.githubusercontent.com/23147681/236591658-c322c310-c222-4b87-8b38-18217930fb54.mp4

**Comment by praydog** [comment]
It's being looked into. I'm trying to do it at the engine level so it's a bit difficult to get inserted into the middle of the pipeline.

**Comment by DaveACG** [comment]
>  Resident.Evil.4.Biohazard.4.2023.05.06.-.03.40.42.02.mp4 
> I also noticed that this issue also appears in close ups even with bloom disabled like here:
> 
>  Resident.Evil.4.Biohazard.4.2023.05.06.-.03.49.40.03.mp4

That's due to the depth of field effect. Basically all post-process effects can exhibit varying levels of artifacting with reconstruction if they're not carefully accounted for, while not the exact same type of artifacting you can see issues in games even with native DLSS support where the developer didn't fully take the implementation of other low-res buffers into account.

**Comment by DaveACG** [comment]
Any possible update on this, or is this looking to be in insurmountable problem?
